File: Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        int min = prices[0];
        
        for(int i = 1; i < prices.size(); i++){
            if(prices[i] < min){
                min = prices[i];
            }
            else{
                ans = ans < prices[i] - min ? prices[i] - min : ans;
            }
        }
        
        return ans;
    }
};

File: Best Time to Buy and Sell Stock.js
var maxProfit = function(prices) {
    const length = prices.length;
    let buy = prices[0];
    let profit = 0;
    
    for(let i = 1; i < length; i++){
        buy = Math.min(buy, prices[i]);
        profit = Math.max(profit, prices[i] - buy);
    }
    
    return profit;
};